app/src/main/java/com/project/squirrelobserver/squirrelObserver/RecordBehaviorTabActivity.java

```diff
@@ -269,8 +278,49 @@ private void hideButtons(String filter, ArrayList<ToggleButton> buttons) {
 
                 button.setVisibility(View.GONE);
             } else {
+                if(isScanMode) {
+                    button.setVisibility(View.VISIBLE);
+                } else {
+                    if(((Behavior) button.getTag()).isAO) {
+                        button.setVisibility(View.VISIBLE);
+                    }
+                }
+            }
+        }
+    }
+
+    public void toggleMode(boolean scanMode) {
+        toggleHideNonAOBehaviors(scanMode, list);
+        toggleHideNonAOBehaviors(scanMode, listFrequent);
+
+//        final GridView gridView = (GridView) findViewById(R.id.behavior_grid_id);
+//        final GridView gridViewRecent = (GridView) findViewById(R.id.behavior_grid_recent_id);
+
+        if (gridView != null && gridViewFrequent != null) {
+            gridView.invalidateViews();
+            gridViewFrequent.invalidateViews();
+        }
+    }
 
-                button.setVisibility(View.VISIBLE);
+    private void toggleHideNonAOBehaviors(boolean scanMode, ArrayList<ToggleButton> buttons) {
+        if(buttons == null)
+            return;
+
+        for(int i=0; i < buttons.size(); i++) {
+            ToggleButton button = buttons.get(i);
+            if (button == null)
+                return;
+
+            Behavior behavior = (Behavior) button.getTag();
+            if(behavior == null)
+                return;
+
+            if (!scanMode) {
```

Comment:
We don't need to check anything if we're in scan mode. Try writing as:

if (!scanMode && !behavior.isAO) {
...View.GONE
} else {
...View.VISIBLE
}